vms/platformvm/state/state.go

```diff
@@ -1672,10 +1672,17 @@ func (s *state) initValidatorSets() error {
 }
 
 func (s *state) write(updateValidators bool, height uint64) error {
+	weightDiffs, blsKeyDiffs, err := s.processCurrentStakers()
+	if err != nil {
+		return err
+	}
+
 	errs := wrappers.Errs{}
 	errs.Add(
 		s.writeBlocks(),
-		s.writeCurrentStakers(updateValidators, height),
+		s.writeCurrentStakers(updateValidators),
+		s.writeWeightDiffs(height, weightDiffs),
+		s.writeBlsKeyDiffs(height, blsKeyDiffs),
```

Comment:
these are some of methods where we only write the information collected in processCurrentStakers